pol-core/plib/uofile00.cpp

```diff
@@ -128,6 +177,121 @@ bool open_uopmap_file( const int mapid, size_t* out_file_size = nullptr )
   return true;
 }
 
+bool open_uopmulti_file( std::map<unsigned int, std::vector<USTRUCT_MULTI_ELEMENT>>& multi_map )
+{
+  std::string filepart = "MultiCollection.uop";
+  std::string filename = systemstate.config.uo_datafile_root + filepart;
+  if ( !Clib::FileExists( filename ) )
+  {
+    INFO_PRINTLN( "{} not found in {}. Searching for old multi.mul/multi.idx files.", filepart,
+                  systemstate.config.uo_datafile_root );
+    return false;
+  }
+
+  std::ifstream ifs;
+  ifs.open( filename, std::ios::binary );
+
+  if ( ifs.fail() )
+    return false;
+
+  kaitai::kstream ks( &ifs );
+  uop_t uopfile( &ks );
+  ByteReader reader;
+  std::unique_ptr<unsigned char[]> uncompressed;
```

Comment:
isn't there a better way to handle the unsigned char[]? like a std::array/std::vector instead of using the unique_ptr?

Reply:
This was just kinda copy-pasted from https://github.com/polserver/polserver/blob/1de9d3a9f60725edf27521a71f91eb581712d484/pol-core/poltool/PolToolMain.cpp#L219-L220